GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -21,27 +24,72 @@ public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool forceRu
         }
 
         public override string Area => nameof(LooseObjectsStep);
+        public int LooseObjectsPutIntoPackFile { get; private set; } = 0;
+        public int MaxLooseObjectsInPack { get; set; } = 100000;
 
         protected override string LastRunTimeFilePath => Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjectsLastRunFileName);
         protected override TimeSpan TimeBetweenRuns => TimeSpan.FromDays(7);
 
         public int CountLooseObjects()
         {
             int count = 0;
-            
+
             foreach (string directoryPath in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
             {
                 string directoryName = directoryPath.TrimEnd(Path.DirectorySeparatorChar).Split(Path.DirectorySeparatorChar).Last();
 
                 if (GitObjects.IsLooseObjectsDirectory(directoryName))
-                {                    
+                {
                     count += this.Context.FileSystem.GetFiles(Path.Combine(this.Context.Enlistment.GitObjectsRoot, directoryPath), "*").Count();
                 }
             }
 
             return count;
         }
 
+        public void PackLooseObjects(StreamWriter streamWriter)
+        {
+            // Find loose Objects
+            foreach (DirectoryItemInfo directoryItemInfo in this.Context.FileSystem.ItemsInDirectory(this.Context.Enlistment.GitObjectsRoot))
+            {
+                if (directoryItemInfo.IsDirectory)
+                {
+                    string directoryName = directoryItemInfo.Name;
+
+                    if (GitObjects.IsLooseObjectsDirectory(directoryName))
+                    {
+                        string[] looseObjectFileNamesInDir = this.Context.FileSystem.GetFiles(Path.Combine(directoryItemInfo.FullName), "*");
+
+                        foreach (string file in looseObjectFileNamesInDir)
+                        {
+                            this.LooseObjectsPutIntoPackFile++;
+                            streamWriter.Write(this.GetLooseObjectId(directoryName, file) + "\n");
+
+                            // Stop when we hit the limit.  The next run will pack more files.
+                            if (this.LooseObjectsPutIntoPackFile == this.MaxLooseObjectsInPack)

```

Comment:
@jrbriggs you had a question here about changes this to <= for batches in teh future.  Not sure what you mean.

Reply:
I prefer to be safe with these stop conditions. If we ever change to adding multiple objects at once, we could skip over `this.MaxLooseObjectsInPack` entirely and pack all loose objects in the cache into one pack. Changing the stop condition to `this.LooseObjectsPutIntoPackFile >= this.MaxLooseObjectsInPack` prevents that. I think I used the wrong operator in the last comment, my apologies if that was the source of confusion.